packages/yoga/src/shared/propTypes/deprecated.js

```diff
@@ -0,0 +1,23 @@
+import warning from 'warning';
+
+let warned = {};
+
+export default function deprecated(propType, explanation) {
+  return function validate(props, propName, componentName, ...rest) {
+    if (props[propName] != null) {
+      const message = `"${propName}" property of "${componentName}" has been deprecated.\n${explanation}`;
+      if (!warned[message]) {
+        warning(false, message);
+        warned[message] = true;
+      }
+    }
+
+    return propType(props, propName, componentName, ...rest);
+  };
+}
+
+function resetWarned() {
+  warned = {};
+}
+
+deprecated.resetWarned = resetWarned;
```

Comment:
Hey man! Nice use of [deprecated prop-type](https://reactjs.org/warnings/dont-call-proptypes.html#fixing-the-false-positive-in-third-party-proptypes)!

I was just thinking here, what's the use for the reset on the yoga context?
can we simplify this implementation?

Like, if we remove the reset there's no need for the `warned` let outside the deprecated function
we could also use the export convention of the other files, creating a `const` and exporting at the end of the file (not really necessary, but just to keep code consistency)


another thing I was looking over here, is that, react-proptypes uses `warning` [as a dependency](https://github.com/react-bootstrap/react-prop-types/blob/master/package.json#L33) but yoga doesn't, is this really working?

ReactJS package is using `console.warn` for [deprecation warning since 15.6.0](https://github.com/reactjs/reactjs.org/blob/a4d4893bd3c56f4f7f821c46ed512eac6c3c21b9/content/blog/2017-06-13-react-v15.6.0.md), can we use it too?

what do you think?

Reply:
> ReactJS package is using console.warn for deprecation warning since 15.6.0, can we use it too?

Agree, updated to use the `console.warn` itself.
And also removed the unnecessary reset function.

Reply:
🚀